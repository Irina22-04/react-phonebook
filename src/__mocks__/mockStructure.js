export const mockStructure = [
    {
        "name": "OOO 'Дети дикой обезьяны'",
        "id": 1
    },
    {
        "name": "Управление 1",
        "parent": 1,
        "id": 2
    },
    {
        "name": "Управление 22",
        "parent": 1,
        "id": 3
    },
    {
        "name": "Управление 32",
        "parent": 1,
        "id": 4
    },
    {
        "name": "Отдел по работе с клиентами",
        "id": 5,
        "parent": 2
    },
    {
        "name": "Отдел по работе с юридическими лицами",
        "id": 8,
        "parent": 2
    }
];