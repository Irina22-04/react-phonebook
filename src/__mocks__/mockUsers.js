export const mockUsers = [
    {
        "name": "Дмитрий",
        "surname": "Дмитриевич",
        "patronym": "Дмитриевич",
        "department": 8,
        "fullname": "Дмитриев Дмитрий Дмитриевич",
        "id": 104,
        "isAdmin": true,
        "email": "admin@mail.ru",
        "password": "$2a$10$AnPxLMUGPKB8eAsdF90pFOWi/h2ejtPhrkzA6.a/nNUty8SexQBQe",
        "position": "специалист",
        "phone": "340-34-34",
        "businessMobile": "(33)333-33-33",
        "personalMobile": "(44)444-44-44",
        "personalEmail": "Dmit@mail.ru",
        "birthday": "22/02/1999"
    },
    {
        "name": "Михаил",
        "surname": "Михайлов",
        "patronym": "Михайлович",
        "department": 14,
        "fullname": "Михайлов Михаил Михайлович",
        "id": 109,
        "position": "специалист",
        "phone": "340-34-34"
    }
];