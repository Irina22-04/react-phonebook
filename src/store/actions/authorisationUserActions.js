import {get} from "lodash";
import jwt_decode from "jwt-decode";

import {getContactById, loginUser} from "../../utils/api";

export const USER_AUTHORISATION = 'USER_AUTHORISATION';
export const LOGIN_TRY = 'LOGIN_TRY';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAIL = 'LOGIN_FAIL';
export const LEAVE_ACCOUNT = 'LEAVE_ACCOUNT';

export function setUserAuthorisation(dispatch, authorisationState) {
    return dispatch({
        type: USER_AUTHORISATION,
        payload: authorisationState,
    })
}

export const loginUserTry = (dispatch, user) => async (dispatch) => {
    dispatch({type: LOGIN_TRY});
    return await loginUser(user)
        .then(({data}) => {
            const {sub} = jwt_decode(data.accessToken);
            return sub
        })
        .then(async (id) => await getContactById(id))
        .then(({data}) => {
            dispatch({
                type: LOGIN_SUCCESS,
                payload: data,
            })
        })
        .catch(e => {
            dispatch({
                type: LOGIN_FAIL,
                payload: get(e, ['response', 'data', 'error_message'], e.message)
            });
            return Promise.reject('Ошибка авторизации')
        });
};

export function leaveAccount(dispatch) {
    return dispatch({
        type: LEAVE_ACCOUNT,
    })
}