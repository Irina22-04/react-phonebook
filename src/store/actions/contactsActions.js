import {get} from "lodash";

import {getAllContacts} from "../../utils/api";

export const CONTACTS_LOADING = 'CONTACTS_LOADING';
export const CONTACTS_SET = 'CONTACTS_SET';
export const CONTACTS_ERROR = 'CONTACTS_ERROR';
export const CONTACTS_SET_PAGE = 'CONTACTS_SET_PAGE';
export const CONTACTS_SET_SURNAME_LETTER = 'CONTACTS_SET_SURNAME_LETTER';
export const CONTACTS_SET_FULLNAME_PATTERN = 'CONTACTS_SET_FULLNAME_PATTERN';
export const RESET_CONTACT_SEARCH = 'RESET_CONTACT_SEARCH';


export function getAllContactsTry(dispatch) {
    return async function () {
        dispatch({type: CONTACTS_LOADING});
        await getAllContacts()
            .then(data => {
                dispatch({
                    type: CONTACTS_SET,
                    payload: data.data,
                })
            })
            .catch(e => {
                dispatch({
                    type: CONTACTS_ERROR,
                    payload: get(e, ['response', 'data', 'error_message'], e.message)
                });
            });
    }
}

export function setPage(dispatch, page) {
    return dispatch({type: CONTACTS_SET_PAGE, payload: page});
}

export function setSurnameLetter(dispatch, letter) {
    return dispatch({type: CONTACTS_SET_SURNAME_LETTER, payload: letter});
}

export function setFullnamePattern(dispatch, pattern) {
    return dispatch({type: CONTACTS_SET_FULLNAME_PATTERN, payload: pattern});
}

export function resetSearchParams(dispatch) {
    return dispatch({type: RESET_CONTACT_SEARCH});
}
