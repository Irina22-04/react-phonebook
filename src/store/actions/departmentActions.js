import {get} from "lodash";

import {addStructure, deleteStructure, getStructure, updateStructure} from "../../utils/api";

export const STRUCTURE_LOADING = 'STRUCTURE_LOADING';
export const STRUCTURE_SET = 'STRUCTURE_SET';
export const STRUCTURE_ERROR = 'STRUCTURE_ERROR';
export const EDIT_DEPARTMENT = 'SET_EDIT_DEPARTMENT';
export const SET_DEPARTMENT = 'SET_DEPARTMENT';
export const STRUCTURE_UPDATE = 'STRUCTURE_UPDATE';
export const STRUCTURE_DELETE = 'STRUCTURE_DELETE';
export const SET_PARENT_DEPARTMENT = 'SET_PARENT_DEPARTMENT';
export const STRUCTURE_ADD = 'STRUCTURE_ADD';
export const CARRY_DEPARTMENT = 'CARRY_DEPARTMENT';
export const SET_DELETE_DEPARTMENT_ERROR = 'SET_DELETE_DEPARTMENT_ERROR';
export const REMOVE_DELETE_DEPARTMENT_ERROR = 'REMOVE_DELETE_DEPARTMENT_ERROR';

export function setDepartment(dispatch, department) {
    return dispatch({
        type: SET_DEPARTMENT,
        payload: department,
    })

}

export function editDepartment(dispatch, department) {
    return dispatch({
        type: EDIT_DEPARTMENT,
        payload: department,
    })

}

export function getStructureTry(dispatch) {
    return async function () {
        dispatch({type: STRUCTURE_LOADING});
        await getStructure()
            .then(data => {
                dispatch({
                    type: STRUCTURE_SET,
                    payload: data.data,
                })
            })
            .catch(e => {
                dispatch({
                    type: STRUCTURE_ERROR,
                    payload: get(e, ['response', 'data', 'error_message'], e.message)
                });
            });
    }
}

export function updateStructureTry(dispatch, department) {
    return async function () {
        await updateStructure(department)
            .then(data => {
                dispatch({
                    type: STRUCTURE_UPDATE,
                    payload: data.data,
                })
            })
            .catch(e => {
                return Promise.reject(e.message);
            });
    }
}

export function deleteStructureTry(dispatch, department) {
    return async function () {
        await deleteStructure(department)
            .then(() => {
                dispatch({
                    type: STRUCTURE_DELETE,
                    payload: department,
                })
            })
            .catch(e => {
                return Promise.reject(e.message);
            });
    }
}

export function addChildrenDepartment(dispatch, parentDepartment) {
    return dispatch({
        type: SET_PARENT_DEPARTMENT,
        payload: parentDepartment,
    })

}

export function addStructureTry(dispatch, department) {
    return async function () {
        await addStructure(department)
            .then((data) => {
                dispatch({
                    type: STRUCTURE_ADD,
                    payload: data.data,
                });
            })
            .catch(e => {
                return Promise.reject(e.message);
            });
    }
}

export function carryDepartment(dispatch, department) {
    return dispatch({
        type: CARRY_DEPARTMENT,
        payload: department,
    })
}

export function deleteDepartmentError(dispatch, error) {
    return dispatch({
        type: SET_DELETE_DEPARTMENT_ERROR,
        payload: error,
    })
}

export function removeDeleteDepartmentError(dispatch) {
    return dispatch({
        type: REMOVE_DELETE_DEPARTMENT_ERROR,
    })
}
