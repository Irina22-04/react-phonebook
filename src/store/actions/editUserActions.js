import {get} from "lodash";
import jwt_decode from "jwt-decode";

import {
    deleteContact,
    addContact,
    getAllContacts,
    editContact,
} from "../../utils/api";


export const USERS_LOADING = 'USERS_LOADING';
export const USERS_SET = 'USERS_SET';
export const USERS_ERROR = 'USERS_ERROR';

export const NEW_CONTACT_LOADING = 'NEW_CONTACT_LOADING';

export const DELETE_CONTACT_LOADING = 'DELETE_CONTACT_LOADING';
export const DELETE_CONTACT_SET = 'DELETE_CONTACT_SET';

export const EDIT_CONTACT_LOADING = 'EDIT_CONTACT_LOADING';
export const EDIT_CONTACT_SET = 'EDIT_CONTACT_SET';

export const NEW_CONTACT_EDIT = 'NEW_CONTACT_EDIT';

export const EDIT_CONTACTS_SET_PAGE = 'EDIT_CONTACTS_SET_PAGE';
export const EDIT_CONTACTS_SET_SEARCH_PATTERN = 'EDIT_CONTACTS_SET_SEARCH_PATTERN';
export const RESET_EDIT_CONTACT_SEARCH = 'RESET_EDIT_CONTACT_SEARCH';

export const CONTACT_ERROR = 'NEW_CONTACT_ERROR';

export function getAllUsersTry(dispatch) {
    return async function () {
        dispatch({type: USERS_LOADING});
        await getAllContacts()
            .then(data => {
                dispatch({
                    type: USERS_SET,
                    payload: data.data,
                })
            })
            .catch(e => {
                dispatch({
                    type: USERS_ERROR,
                    payload: get(e, ['response', 'data', 'error_message'], e.message)
                });
            });
    }
}

export function addNewContactTry(dispatch, contact) {
    return async function () {
        dispatch({type: NEW_CONTACT_LOADING});
        await addContact(contact)
            .then((data) => {
                const {sub} = jwt_decode(data.data.accessToken);
                return sub;
            })
            .then((data) => {
                dispatch({
                    type: NEW_CONTACT_EDIT,
                    payload: data,
                })
            })
            .catch(e => {
                dispatch({
                    type: CONTACT_ERROR,
                    payload: get(e, ['response', 'data', 'error_message'], e.message)
                });
                return Promise.reject(e.message);
            });
    }
}

export function deleteContactTry(dispatch, contactId) {
    return async function () {
        dispatch({type: DELETE_CONTACT_LOADING});
        await deleteContact(contactId)
            .then(() => {
                dispatch({
                    type: DELETE_CONTACT_SET,
                    payload: contactId,
                })
            })
            .catch(e => {
                dispatch({
                    type: CONTACT_ERROR,
                    payload: get(e, ['response', 'data', 'error_message'], e.message)
                });
                return Promise.reject(e.message);
            });
    }
}

export function editContactTry(dispatch, contact) {
    return async function () {
        dispatch({type: EDIT_CONTACT_LOADING});
        const {password, ...withoutPassword} = contact;
        const editedContact = password ? contact : withoutPassword;
        return await editContact(editedContact)
            .then(() => {
                dispatch({
                    type: EDIT_CONTACT_SET,
                    payload: contact,
                });
            })
            .catch(e => {
                dispatch({
                    type: CONTACT_ERROR,
                    payload: get(e, ['response', 'data', 'error_message'], e.message)
                });
                return Promise.reject(e.message);
            });
    };
}

export function setPage(dispatch, page){
    return dispatch({type: EDIT_CONTACTS_SET_PAGE, payload: page});
}

export function setSearchPattern(dispatch, pattern){
    return dispatch({type: EDIT_CONTACTS_SET_SEARCH_PATTERN, payload: pattern});
}

export function resetSearchParams(dispatch) {
    return dispatch({type: RESET_EDIT_CONTACT_SEARCH});
}

