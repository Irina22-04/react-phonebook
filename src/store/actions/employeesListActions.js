import {get} from "lodash";

import {getEmployeesByStructure} from "../../utils/api";

export const EMPLOYEES_LOADING = 'EMPLOYEES_LOADING';
export const EMPLOYEES_SET = 'EMPLOYEES_SET';
export const EMPLOYEES_ERROR = 'EMPLOYEES_ERROR';

export function getEmployeesByStructureTry(dispatch, department) {
    return async function () {
        dispatch({type: EMPLOYEES_LOADING});
        await getEmployeesByStructure(department)
            .then(data => {
                dispatch({
                    type: EMPLOYEES_SET,
                    payload: data.data,
                })
            })
            .catch(e => {
                dispatch({
                    type: EMPLOYEES_ERROR,
                    payload: get(e, ['response', 'data', 'error_message'], e.message)
                });
            });
    }
}
