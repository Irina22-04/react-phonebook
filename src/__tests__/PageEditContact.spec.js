import React from "react";
import renderer, {act} from "react-test-renderer";
import {Provider} from "react-redux";
import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";
import {BrowserRouter} from "react-router-dom";

import {mockUsers} from "../__mocks__/mockUsers";
import PageEditContact from "../containers/PageEditContacts/PageEditContacts";
import {USERS_LOADING} from "../store/actions/editUserActions";


const initialState = {
    status: 3,
    usersList: mockUsers,
    error: '',
    updateStatus: 0,
    addedContact: '',
    paginationOptions: {
        page: 1,
        limit: 15
    },
    searchOptions: ''
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


describe('PageEditContact', () => {
    const state = {usersListForEdit: initialState};
    const store = mockStore(() => state);
    const component = renderer.create(<Provider store={store}><BrowserRouter>
        <PageEditContact location={{search: null}} history={[]}/></BrowserRouter></Provider>);

    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    it('dispatch USERS_LOADING after render', () => {
        expect(store.getActions()).toEqual([{type: USERS_LOADING}]);
    });

    it('button edit user should work', () => {

        const buttons = component.root.findAllByType('button');
        buttons[4].props.onClick();

        let componentTree = component.toJSON();
        expect(componentTree).toMatchSnapshot();
    });

    it('button add user should work', () => {

        const button = component.root.findByProps({className: "all-users-button edit-user-page-button"});
        button.props.onClick();

        let componentTree = component.toJSON();
        expect(componentTree).toMatchSnapshot();
    });

});

