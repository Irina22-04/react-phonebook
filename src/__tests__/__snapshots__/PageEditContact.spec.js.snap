// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="page-container"
  onClick={[Function]}
>
  <div
    className="search-field"
  >
    <div
      className="search-field"
    >
      <form
        onSubmit={[Function]}
      >
        <div>
          <div
            className="search-name"
          >
            Поиск контакта
          </div>
          <div
            className="ui fluid left icon input search-input"
          >
            <input
              disabled={false}
              maxLength="40"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Введите имя"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="users icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="ui buttons search-contacts-buttons"
        >
          <button
            className="ui disabled button"
            disabled={true}
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            Очистить
          </button>
          <div
            className="or"
            data-text=""
          />
          <button
            className="ui positive disabled button"
            disabled={true}
            onClick={[Function]}
            tabIndex={-1}
            type="submit"
          >
            Искать
          </button>
        </div>
      </form>
    </div>
  </div>
  <button
    className="ui small button all-users-button"
    onClick={[Function]}
  >
    Все
  </button>
  <div
    className="edit-user-page"
  >
    <div>
      <div
        className="contact-wrapper"
        id={104}
      >
        <div
          className="user-name"
        >
          Дмитриевич Дмитрий Дмитриевич
        </div>
        <div
          className="edit-user-buttons"
        >
          <div
            className="ui buttons"
          >
            <button
              className="ui button"
              id={104}
              onClick={[Function]}
              type="button"
            >
              Удалить
            </button>
            <div
              className="or"
              data-text=""
            />
            <a
              href="/editUser/104"
              onClick={[Function]}
            >
              <button
                className="ui positive button"
                onClick={[Function]}
                type="button"
              >
                Редактировать
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        className="contact-wrapper"
        id={109}
      >
        <div
          className="user-name"
        >
          Михайлов Михаил Михайлович
        </div>
        <div
          className="edit-user-buttons"
        >
          <div
            className="ui buttons"
          >
            <button
              className="ui button"
              id={109}
              onClick={[Function]}
              type="button"
            >
              Удалить
            </button>
            <div
              className="or"
              data-text=""
            />
            <a
              href="/editUser/109"
              onClick={[Function]}
            >
              <button
                className="ui positive button"
                onClick={[Function]}
                type="button"
              >
                Редактировать
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="ui small button all-users-button edit-user-page-button"
    onClick={[Function]}
  >
    Добавить
  </button>
  <div
    aria-label="Pagination Navigation"
    className="ui pagination menu"
    role="navigation"
  >
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="First item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="firstItem"
      value={1}
    >
      «
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Previous item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="prevItem"
      value={1}
    >
      ⟨
    </a>
    <a
      aria-current={true}
      aria-disabled={false}
      className="active item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="pageItem"
      value={1}
    >
      1
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Next item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="nextItem"
      value={1}
    >
      ⟩
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Last item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="lastItem"
      value={1}
    >
      »
    </a>
  </div>
</div>
`;

exports[`PageEditContact button add user should work 1`] = `
<div
  className="page-container"
  onClick={[Function]}
>
  <div
    className="search-field"
  >
    <div
      className="search-field"
    >
      <form
        onSubmit={[Function]}
      >
        <div>
          <div
            className="search-name"
          >
            Поиск контакта
          </div>
          <div
            className="ui fluid left icon input search-input"
          >
            <input
              disabled={false}
              maxLength="40"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Введите имя"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="users icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="ui buttons search-contacts-buttons"
        >
          <button
            className="ui disabled button"
            disabled={true}
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            Очистить
          </button>
          <div
            className="or"
            data-text=""
          />
          <button
            className="ui positive disabled button"
            disabled={true}
            onClick={[Function]}
            tabIndex={-1}
            type="submit"
          >
            Искать
          </button>
        </div>
      </form>
    </div>
  </div>
  <button
    className="ui small button all-users-button"
    onClick={[Function]}
  >
    Все
  </button>
  <div
    className="edit-user-page"
  >
    <div>
      <div
        className="contact-wrapper"
        id={104}
      >
        <div
          className="user-name"
        >
          Дмитриевич Дмитрий Дмитриевич
        </div>
        <div
          className="edit-user-buttons"
        >
          <div
            className="ui buttons"
          >
            <button
              className="ui button"
              id={104}
              onClick={[Function]}
              type="button"
            >
              Удалить
            </button>
            <div
              className="or"
              data-text=""
            />
            <a
              href="/editUser/104"
              onClick={[Function]}
            >
              <button
                className="ui positive button"
                onClick={[Function]}
                type="button"
              >
                Редактировать
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        className="contact-wrapper"
        id={109}
      >
        <div
          className="user-name"
        >
          Михайлов Михаил Михайлович
        </div>
        <div
          className="edit-user-buttons"
        >
          <div
            className="ui buttons"
          >
            <button
              className="ui button"
              id={109}
              onClick={[Function]}
              type="button"
            >
              Удалить
            </button>
            <div
              className="or"
              data-text=""
            />
            <a
              href="/editUser/109"
              onClick={[Function]}
            >
              <button
                className="ui positive button"
                onClick={[Function]}
                type="button"
              >
                Редактировать
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="ui small disabled button all-users-button edit-user-page-button"
    disabled={true}
    onClick={[Function]}
    tabIndex={-1}
  >
    Добавить
  </button>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui inverted loader"
      />
    </div>
  </div>
  <div
    aria-label="Pagination Navigation"
    className="ui pagination menu"
    role="navigation"
  >
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="First item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="firstItem"
      value={1}
    >
      «
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Previous item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="prevItem"
      value={1}
    >
      ⟨
    </a>
    <a
      aria-current={true}
      aria-disabled={false}
      className="active item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="pageItem"
      value={1}
    >
      1
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Next item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="nextItem"
      value={1}
    >
      ⟩
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Last item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="lastItem"
      value={1}
    >
      »
    </a>
  </div>
</div>
`;

exports[`PageEditContact button edit user should work 1`] = `
<div
  className="page-container"
  onClick={[Function]}
>
  <div
    className="search-field"
  >
    <div
      className="search-field"
    >
      <form
        onSubmit={[Function]}
      >
        <div>
          <div
            className="search-name"
          >
            Поиск контакта
          </div>
          <div
            className="ui fluid left icon input search-input"
          >
            <input
              disabled={false}
              maxLength="40"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Введите имя"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="users icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="ui buttons search-contacts-buttons"
        >
          <button
            className="ui disabled button"
            disabled={true}
            onClick={[Function]}
            tabIndex={-1}
            type="button"
          >
            Очистить
          </button>
          <div
            className="or"
            data-text=""
          />
          <button
            className="ui positive disabled button"
            disabled={true}
            onClick={[Function]}
            tabIndex={-1}
            type="submit"
          >
            Искать
          </button>
        </div>
      </form>
    </div>
  </div>
  <button
    className="ui small button all-users-button"
    onClick={[Function]}
  >
    Все
  </button>
  <div
    className="edit-user-page"
  >
    <div>
      <div
        className="contact-wrapper"
        id={104}
      >
        <div
          className="user-name"
        >
          Дмитриевич Дмитрий Дмитриевич
        </div>
        <div
          className="edit-user-buttons"
        >
          <div
            className="ui buttons"
          >
            <button
              className="ui button"
              id={104}
              onClick={[Function]}
              type="button"
            >
              Удалить
            </button>
            <div
              className="or"
              data-text=""
            />
            <a
              href="/editUser/104"
              onClick={[Function]}
            >
              <button
                className="ui positive button"
                onClick={[Function]}
                type="button"
              >
                Редактировать
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        className="contact-wrapper"
        id={109}
      >
        <div
          className="user-name"
        >
          Михайлов Михаил Михайлович
        </div>
        <div
          className="edit-user-buttons"
        >
          <div
            className="ui buttons"
          >
            <button
              className="ui button"
              id={109}
              onClick={[Function]}
              type="button"
            >
              Удалить
            </button>
            <div
              className="or"
              data-text=""
            />
            <a
              href="/editUser/109"
              onClick={[Function]}
            >
              <button
                className="ui positive button"
                onClick={[Function]}
                type="button"
              >
                Редактировать
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="ui small button all-users-button edit-user-page-button"
    onClick={[Function]}
  >
    Добавить
  </button>
  <div
    aria-label="Pagination Navigation"
    className="ui pagination menu"
    role="navigation"
  >
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="First item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="firstItem"
      value={1}
    >
      «
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Previous item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="prevItem"
      value={1}
    >
      ⟨
    </a>
    <a
      aria-current={true}
      aria-disabled={false}
      className="active item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="pageItem"
      value={1}
    >
      1
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Next item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="nextItem"
      value={1}
    >
      ⟩
    </a>
    <a
      aria-current={false}
      aria-disabled={false}
      aria-label="Last item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="lastItem"
      value={1}
    >
      »
    </a>
  </div>
</div>
`;
